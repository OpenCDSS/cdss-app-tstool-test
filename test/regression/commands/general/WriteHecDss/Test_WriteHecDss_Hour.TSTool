# Write Hourly data to DSS
#@os Windows
# Test setup...
StartLog(LogFile="Results/Test_WriteHecDss_Hour.TSTool.log")
# SetDebugLevel(LogFileLevel=10)
# Write a single blank time series to the dss file to allow WriteHecDss to be closed; allows deleting below.
TS Blank = NewTimeSeries(NewTSID="Blank...Month",SetStart="1960-01",SetEnd="1960-12",Units="CFS")
WriteHecDss(OutputFile="Results\Test_WriteHecDss_Hour_out.dss",Type=PER-AVER,Close=True)
Free(TSList=AllMatchingTSID,TSID="*")
RemoveFile(InputFile="Results\Test_WriteHecDss_Hour_out.dss")
RemoveFile(InputFile="Results\Test_WriteHecDss_Hour.dssutl.out",IfNotFound=Ignore)
# The generated time series has identifier parts that indicate the HEC-DSS
# parts, other than the D part, which is related to the period of record.
TS Alias1 = NewPatternTimeSeries(NewTSID="Apart1:Bpart1-with-dashes..Cpart1.Hour.Fpart1",Description="Test data",SetStart="1950-01-01 00",SetEnd="1950-02-11 13",Units="CFS",PatternValues="5,10,12,13,75")
TS Alias2 = NewPatternTimeSeries(NewTSID="Apart1:Bpart1..Cpart1.Hour.Start on hour 24",Description="Start on hour 24",SetStart="1950-01-01 24",SetEnd="1950-01-11 13",Units="CFS",PatternValues="5,10,12,13,75")
TS Alias3 = NewPatternTimeSeries(NewTSID="Apart1:Bpart1..Cpart1.6Hour.6 hour data",Description="6 hour data",SetStart="1950-01-01 24",SetEnd="1950-01-11 12",Units="CFS",PatternValues="5,10,12,13,75")
TS Alias4 = NewPatternTimeSeries(NewTSID="Apart1:Bpart1..Cpart1.4Hour.4 hour data",Description="4 hour data",SetStart="1950-01-01 24",SetEnd="1950-01-11 12",Units="CFS",PatternValues="5,10,12,13,75")
WriteHecDss(OutputFile="Results/Test_WriteHecDss_Hour_out.dss",Type=INST-VAL)
#
# Use DSSUTL to output the data from the resulting DSS file to compare results
RunDSSUTL(DssFile="Results\Test_WriteHecDss_Hour_out.dss",InputFile="Test_WriteHecDss_Hour.MCO",DssutlProgram="${WorkingDir}\DSSUTL.EXE")
# In order to regenerate the expected results file, edit the output file in the file: Test_WriteHecDss_Hour.MCO
CompareFiles(InputFile1="ExpectedResults\Test_WriteHecDss_Hour.dssutl.out",InputFile2="Results/Test_WriteHecDss_Hour.dssutl.out",WarnIfDifferent=True)
