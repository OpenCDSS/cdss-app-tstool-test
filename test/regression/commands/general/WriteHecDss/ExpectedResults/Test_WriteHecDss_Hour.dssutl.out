  
 /APART1/BPART1/CPART1/01JAN1950/4HOUR/4 HOUR DATA/
 Start: 01JAN1950 at 2400 hours;   End: 11JAN1950 at 1200 hours;  Number:   58
 Units: CFS        Type: INST-VAL
 01JAN1950, 2400;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 03JAN1950, 0400;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 04JAN1950, 0800;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 05JAN1950, 1200;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 06JAN1950, 1600;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 07JAN1950, 2000;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 08JAN1950, 2400;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 10JAN1950, 0400;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 11JAN1950, 0800;    10.00   12.00
 ----------------------------------------------------------------------------
  
 /APART1/BPART1/CPART1/01JAN1950/6HOUR/6 HOUR DATA/
 Start: 01JAN1950 at 2400 hours;   End: 11JAN1950 at 1200 hours;  Number:   39
 Units: CFS        Type: INST-VAL
 01JAN1950, 2400;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 03JAN1950, 1800;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 05JAN1950, 1200;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 07JAN1950, 0600;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 08JAN1950, 2400;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 10JAN1950, 1800;     5.00   10.00   12.00   13.00
 ----------------------------------------------------------------------------
  
 /APART1/BPART1/CPART1/01JAN1950/1HOUR/START ON HOUR 24/
 Start: 01JAN1950 at 2400 hours;   End: 11JAN1950 at 1300 hours;  Number:  230
 Units: CFS        Type: INST-VAL
 01JAN1950, 2400;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 02JAN1950, 0700;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 02JAN1950, 1400;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 02JAN1950, 2100;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 03JAN1950, 0400;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 03JAN1950, 1100;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 03JAN1950, 1800;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 04JAN1950, 0100;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 04JAN1950, 0800;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 04JAN1950, 1500;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 04JAN1950, 2200;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 05JAN1950, 0500;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 05JAN1950, 1200;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 05JAN1950, 1900;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 06JAN1950, 0200;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 06JAN1950, 0900;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 06JAN1950, 1600;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 06JAN1950, 2300;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 07JAN1950, 0600;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 07JAN1950, 1300;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 07JAN1950, 2000;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 08JAN1950, 0300;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 08JAN1950, 1000;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 08JAN1950, 1700;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 08JAN1950, 2400;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 09JAN1950, 0700;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 09JAN1950, 1400;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 09JAN1950, 2100;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 10JAN1950, 0400;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 10JAN1950, 1100;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 10JAN1950, 1800;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 11JAN1950, 0100;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 11JAN1950, 0800;    75.00    5.00   10.00   12.00   13.00   75.00
 ----------------------------------------------------------------------------
  
 /APART1/BPART1-WITH-DASHES/CPART1/01DEC1949/1HOUR/FPART1/
 Start: 31DEC1949 at 2400 hours;   End: 31DEC1949 at 2400 hours;  Number:    1
 Units: CFS        Type: INST-VAL
 31DEC1949, 2400;     5.00
 ----------------------------------------------------------------------------
  
 /APART1/BPART1-WITH-DASHES/CPART1/01JAN1950/1HOUR/FPART1/
 Start: 01JAN1950 at 0100 hours;   End: 31JAN1950 at 2400 hours;  Number:  744
 Units: CFS        Type: INST-VAL
 01JAN1950, 0100;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 01JAN1950, 0800;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 01JAN1950, 1500;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 01JAN1950, 2200;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 02JAN1950, 0500;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 02JAN1950, 1200;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 02JAN1950, 1900;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 03JAN1950, 0200;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 03JAN1950, 0900;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 03JAN1950, 1600;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 03JAN1950, 2300;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 04JAN1950, 0600;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 04JAN1950, 1300;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 04JAN1950, 2000;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 05JAN1950, 0300;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 05JAN1950, 1000;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 05JAN1950, 1700;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 05JAN1950, 2400;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 06JAN1950, 0700;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 06JAN1950, 1400;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 06JAN1950, 2100;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 07JAN1950, 0400;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 07JAN1950, 1100;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 07JAN1950, 1800;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 08JAN1950, 0100;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 08JAN1950, 0800;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 08JAN1950, 1500;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 08JAN1950, 2200;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 09JAN1950, 0500;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 09JAN1950, 1200;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 09JAN1950, 1900;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 10JAN1950, 0200;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 10JAN1950, 0900;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 10JAN1950, 1600;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 10JAN1950, 2300;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 11JAN1950, 0600;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 11JAN1950, 1300;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 11JAN1950, 2000;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 12JAN1950, 0300;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 12JAN1950, 1000;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 12JAN1950, 1700;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 12JAN1950, 2400;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 13JAN1950, 0700;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 13JAN1950, 1400;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 13JAN1950, 2100;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 14JAN1950, 0400;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 14JAN1950, 1100;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 14JAN1950, 1800;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 15JAN1950, 0100;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 15JAN1950, 0800;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 15JAN1950, 1500;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 15JAN1950, 2200;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 16JAN1950, 0500;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 16JAN1950, 1200;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 16JAN1950, 1900;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 17JAN1950, 0200;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 17JAN1950, 0900;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 17JAN1950, 1600;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 17JAN1950, 2300;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 18JAN1950, 0600;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 18JAN1950, 1300;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 18JAN1950, 2000;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 19JAN1950, 0300;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 19JAN1950, 1000;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 19JAN1950, 1700;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 19JAN1950, 2400;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 20JAN1950, 0700;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 20JAN1950, 1400;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 20JAN1950, 2100;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 21JAN1950, 0400;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 21JAN1950, 1100;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 21JAN1950, 1800;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 22JAN1950, 0100;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 22JAN1950, 0800;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 22JAN1950, 1500;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 22JAN1950, 2200;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 23JAN1950, 0500;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 23JAN1950, 1200;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 23JAN1950, 1900;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 24JAN1950, 0200;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 24JAN1950, 0900;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 24JAN1950, 1600;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 24JAN1950, 2300;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 25JAN1950, 0600;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 25JAN1950, 1300;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 25JAN1950, 2000;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 26JAN1950, 0300;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 26JAN1950, 1000;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 26JAN1950, 1700;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 26JAN1950, 2400;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 27JAN1950, 0700;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 27JAN1950, 1400;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 27JAN1950, 2100;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 28JAN1950, 0400;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 28JAN1950, 1100;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 28JAN1950, 1800;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 29JAN1950, 0100;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 29JAN1950, 0800;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 29JAN1950, 1500;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 29JAN1950, 2200;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 30JAN1950, 0500;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 30JAN1950, 1200;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 30JAN1950, 1900;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 31JAN1950, 0200;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 31JAN1950, 0900;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 31JAN1950, 1600;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 31JAN1950, 2300;    13.00   75.00
 ----------------------------------------------------------------------------
  
 /APART1/BPART1-WITH-DASHES/CPART1/01FEB1950/1HOUR/FPART1/
 Start: 01FEB1950 at 0100 hours;   End: 11FEB1950 at 1300 hours;  Number:  253
 Units: CFS        Type: INST-VAL
 01FEB1950, 0100;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 01FEB1950, 0800;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 01FEB1950, 1500;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 01FEB1950, 2200;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 02FEB1950, 0500;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 02FEB1950, 1200;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 02FEB1950, 1900;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 03FEB1950, 0200;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 03FEB1950, 0900;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 03FEB1950, 1600;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 03FEB1950, 2300;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 04FEB1950, 0600;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 04FEB1950, 1300;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 04FEB1950, 2000;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 05FEB1950, 0300;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 05FEB1950, 1000;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 05FEB1950, 1700;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 05FEB1950, 2400;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 06FEB1950, 0700;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 06FEB1950, 1400;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 06FEB1950, 2100;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 07FEB1950, 0400;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 07FEB1950, 1100;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 07FEB1950, 1800;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 08FEB1950, 0100;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 08FEB1950, 0800;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 08FEB1950, 1500;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 08FEB1950, 2200;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 09FEB1950, 0500;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 09FEB1950, 1200;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 09FEB1950, 1900;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 10FEB1950, 0200;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 10FEB1950, 0900;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 10FEB1950, 1600;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 10FEB1950, 2300;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 11FEB1950, 0600;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 11FEB1950, 1300;    12.00
 ----------------------------------------------------------------------------
  
